### PR TITLE
Optimized processing for 1d strided arrays

### PR DESCRIPTION
**Pull request by magnus-nomono**
With some processing of the strides this should be possible to expand to higher dimensional arrays that can be simplified down to 1 stride, but currently only support 1d arrays.

Diplib is still faster for the not just flipped case, but that is because they do multithreading (hardcoded, so we can't disable for testing)

### CODE REVIEW

**Review by iver56 — APPROVED** [review]

File: numpy_minmax/_minmax.c [hunk 1]
```diff
@@ -63,22 +60,118 @@ MinMaxResult minmax_avx2(float *a, size_t length) {
...
+    for (; i <= (length - 8)*stride; i += 8*stride) {
+        __m256 vals = _mm256_set_ps(
+            *(float*)(a + i),
+            *(float*)(a + i + stride),
+            *(float*)(a + i + 2*stride),
+            *(float*)(a + i + 3*stride),
+            *(float*)(a + i + 4*stride),
+            *(float*)(a + i + 5*stride),
+            *(float*)(a + i + 6*stride),
+            *(float*)(a + i + 7*stride)
```
Comment (iver56) [inline numpy_minmax/_minmax.c:140]: Is there an AVX instruction that can do these multiply-accumulate operations?

Comment (magnus-nomono) [reply numpy_minmax/_minmax.c:140]: There is an AVX instruction for multiply-accumulate, but that is not what we are doing here, did you mean something else?

Comment (iver56) [reply numpy_minmax/_minmax.c:140]: Yeah, sorry, I misread the code

Comment (magnus-nomono) [reply numpy_minmax/_minmax.c:140]: Or did you mean the indices? Do you think that is worth it?

Comment (magnus-nomono) [reply numpy_minmax/_minmax.c:140]: Tested with `_mm256_i32gather_ps` (vgatherdps), and it was slower on my computer.

**Pull request merged**